core/include/kernel/rpc_io_i2c.h

```diff
@@ -0,0 +1,32 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2020 Foundries Ltd <jorge@foundries.io>
+ */
+
+#ifndef __RPC_IO_I2C_H
+#define __RPC_IO_I2C_H
+
+#include <tee_api_types.h>
+
+enum rpc_i2c_mode {
+	RPC_I2C_MODE_WRITE = 0,
+	RPC_I2C_MODE_READ = 1,
+};
+
+/*
+ * The bus identifier defines an implicit ABI with the REE.
+ * Using this service to access I2C chips on REE dynamically assigned buses is
+ * not recommended due to the lack of guarantees that the REE will reuse the
+ * same bus identifier over reboots.
+ */
+struct rpc_i2c_request {
+	enum rpc_i2c_mode mode;
+	uint8_t bus; /* bus identifier used by the REE [0..n] */
+	uint8_t chip; /* chip identifier from the device data sheet [0..0x7f] */
```

Comment:
Not considering 10 bit slave address support?
LGTM but should maybe be stated in rpc_io_i2c.h.

Reply:
I didnt need it. I suppose that if this ends up being merged  (not so sure anymore) and someone has a need and a device to test it should be easy to add.

Reply:
How will 10 bit slave address support affect the ABI?

Reply:
On HW, this is a I2C protocol matter in the transfered data.
On linux side, 10bit or 7bit address is an attribute of the I2C client driver. Adding 10bit address support needs a `u16` for `bus` and an attribute stating the slave I2C address size.

Reply:
right, it is not about the technical complexity of the implementation really  but I have no means of testing  it with an actual device. if that is ok with you - accept that I wont be testing access to those chips- I am ok increasing the size for bus and chip to unsigned short for the chip and unsigned int for the bus to match linux sizes. do you prefer if I do that?

Reply:
So no impact on the ABI since we're transmitting those as `uint32_t`.

Reply:
right, it would be just on this side of the world

Reply:
fine with me then as for the field types.